src/python/pants/backend/python/target_types.py

```diff
@@ -54,19 +54,41 @@ class PythonSources(Sources):
     expected_file_extensions = (".py", ".pyi")
 
 
-class PythonInterpreterCompatibility(StringOrStringSequenceField):
-    """A string for Python interpreter constraints on this target.
+class InterpreterConstraintsField(StringSequenceField):
+    """The Python interpreters this code is compatible with.
+
+    Each element should be written in pip-style format, e.g. 'CPython==2.7.*' or 'CPython>=3.6,<4'.
+    You can leave off `CPython` as a shorthand, e.g. '>=2.7' will be expanded to 'CPython>=2.7'.
 
-    This should be written in Requirement-style format, e.g. `CPython==2.7.*` or `CPython>=3.6,<4`.
-    As a shortcut, you can leave off `CPython`, e.g. `>=2.7` will be expanded to `CPython>=2.7`.
+    Specify more than one element to OR the constraints, e.g. `['PyPy==3.7.*', 'CPython==3.7.*']`
+    means either PyPy 3.7 _or_ CPython 3.7.
 
-    If this is left off, this will default to the option `interpreter_constraints` in the
-    [python-setup] scope.
+    If the field is not set, it will default to the option
+    `[python-setup].interpreter_constraints]`.
 
     See https://www.pantsbuild.org/docs/python-interpreter-compatibility.
```

Comment:
I'd like to update this link to be `interpreter-constraints`, but probably not worth breaking the URL.